params/config.go

```diff
@@ -211,6 +213,53 @@ func (c ChainConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(raw)
 }
 
+type ChainConfigWithUpgradesMarshalled struct {
+	*ChainConfig
+	UpgradeConfig UpgradeConfig `json:"upgrades,omitempty"`
+}
+
+// MarshalJSON implements json.Marshaler. This is a workaround for the fact that
+// the embedded ChainConfig struct has a MarshalJSON method, which prevents
+// the default JSON marshalling from working for UpgradeConfig.
+// TODO: consider removing this method by allowing external tag for the embedded
+// ChainConfig struct.
+func (s *ChainConfigWithUpgradesMarshalled) MarshalJSON() ([]byte, error) {
```

Comment:
Do we need a custom unmarshaler too for them to use?

Reply:
Yes, this may need to be a follow on to get this bug fix in for the release today.